src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -857,7 +857,7 @@ private void ProcessMessageCleanup(MessageRpc rpc)
                 {
                     try
                     {
-                        rpc.Channel.DecrementActivity();
+                        rpc.Channel.DecrementActivity().GetAwaiter().GetResult();
                     }
```

Comment:
This method is called from ProcessError which is async so you can make this method async and await here.

Reply:
Yes, I saw that but I wasn't sure how far I should go. This method calls some additional `Close` methods which than can be made async as well.

Reply:
I went ahead and removed some more cases of sync over async in `ProcessMessageCleanup`. Please check if these changes should be incldued.